### PR TITLE
UPSTREAM: fix emptyDir idempotency bug

### PR DESCRIPTION
**Pull request by pmorie**
Fix for #3280 

@smarterclayton @ncdc

### CODE REVIEW

**Comment by ncdc** [comment]
LGTM

**Comment by pmorie** [comment]
I tested by running e2e and checking for empty_dir log messages:

```
$ grep empty_dir.go /tmp/openshift-e2e/logs/openshift.log | cut -b 49- | wc -l
49

★ pmorie:~/code/origin (emptydir-security-context) 
$ grep empty_dir.go /tmp/openshift-e2e/logs/openshift.log | cut -b 49- | uniq | wc -l
49
```

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go [hunk 1]
```diff
@@ -203,7 +203,7 @@ func (ed *emptyDir) SetUpAt(dir string) error {
 		err = fmt.Errorf("unknown storage medium %q", ed.medium)
 	}
 
-	if err != nil {
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go]: change to this?
```
if err != nil {
  return err
}

volumeutil.SetReady(ed.getMetaDir())

return nil
```

**Comment by smarterclayton** [comment]
[test]

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go (hunk 1 above)
Comment (pmorie) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go]: I will in a follow up

**Comment by smarterclayton** [comment]
[merge]

**Comment by pmorie** [comment]
For: https://github.com/GoogleCloudPlatform/kubernetes/pull/9844

**Pull request merged**